### DISCUSSION TITLE
Get repeated advertisements from the same device

### DISCUSSION THREAD

**Discussion by christakahashi** [hbldh/bleak#1594] [Q&A] [answered]
I have a device that advertises about every 100ms.  Once a second it updates a byte of service data in that advertisement.  BleakScanner(callback)  seems to only call callback on that device once every (roughly) 10 seconds.  The documentation says that callback is only called on new or changed advertisements.  If data counts as a change I'd expect it to get called every second.  If data doesn't count as a change I'd expect it to only get called once.  Either way the behavior is unexpected (to me).  What I'd like is the former, or failing that, a call back for every advertisement.  Is there any way to configure BleakScanner to do this?  Here's the code I'm using for testing:

```
import asyncio
from bleak import BleakScanner

async def main():
    stop_event = asyncio.Event()

    def callback(device, advertising_data):
        print(device)
        print(advertising_data)

    async with BleakScanner(callback,service_uuids=["<my service uuid>"]) as scanner:
        await stop_event.wait()


asyncio.run(main())
```

**Comment by dlech** [comment]
What operating system and operating system version are you using?

**Reply by christakahashi** [reply]
its a raspberry pi 4.
```
 Operating System: Raspbian GNU/Linux 10 (buster)
 Kernel: Linux 5.10.17-v7l+
 Architecture: arm
```
not sure if relevant but i have bluez installed:
```
bluez is already the newest version (5.50-1.2~deb10u5).
```

**Reply by dlech** [reply, marked as answer]
The problem could be because you are using a rather old kernel and BlueZ. These haven't been tested with Bleak in several years.

**Reply by christakahashi** [reply]
Ah okay.  I'll update and try again. ~is blueZ required or should i remove it?~ (never mind that question. the readme says `Supports Linux distributions with BlueZ` )

**Reply by christakahashi** [reply]
fresh install of rpi OS 12 64-bit, bluez 0.66, and bleak 0.22  seems to be working as expected now.  Thanks!
